x/ibc/09-localhost/types/client_state.go

```diff
@@ -91,13 +91,36 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 	return nil, sdkerrors.Wrap(clienttypes.ErrInvalidEvidence, "cannot submit misbehaviour to localhost client")
 }
 
-// VerifyClientConsensusState returns an error since a local host client does not store consensus
+// VerifyClientState verifies that the localhost client state is stored locally
+func (cs ClientState) VerifyClientState(
+	store sdk.KVStore, cdc codec.BinaryMarshaler, _ commitmentexported.Root,
+	_ uint64, _ commitmentexported.Prefix, _ string, _ []byte, clientState clientexported.ClientState,
+) error {
+	path := host.KeyClientState()
+	bz := store.Get(path)
+	if bz == nil {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedClientStateVerification,
+			"not found for path: %s", path)
+	}
+
+	prevClient := clienttypes.MustUnmarshalClientState(cdc, bz)
+
+	if !reflect.DeepEqual(prevClient, clientState) {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedClientStateVerification,
+			"stored clientState != provided clientState: \n%v\n≠\n%v",
+			prevClient, clientState,
+		)
+	}
+	return nil
+}
+
+// VerifyClientConsensusState returns nil since a local host client does not store consensus
 // states.
```

Comment:
this is actually not intuitive to me and is also not documented on the spec if it should return an error or not. cc @cwgoes

Reply:
If this returns an error, then you can never open a connection with localhost client because the `VerifyClientConsensusState` will fail.

Reply:
ditto on adding to localhost spec dir

Reply:
It should not return an error; it should just pass.